src/common/src/config.rs

```diff
@@ -348,6 +348,31 @@ impl Default for StorageConfig {
     }
 }
 
+impl StorageConfig {
+    /// Checks whether an embedded compactor starts with a compute node.
+    #[inline(always)]
+    pub fn embedded_compactor_enabled(&self) -> bool {
+        // We treat `hummock+memory-shared` as a shared storage, so we won't start the compactor
+        // along with the compute node.
+        self.state_store == "hummock+memory"
+            || self.state_store.starts_with("hummock+disk")
+            || self.disable_remote_compactor
+    }
+
+    /// The maximal memory that storage components may use based on the configurations.
+    pub fn total_memory_limit_mb(&self) -> usize {
```

Comment:
How about rename it to `total_storage_memory_limit_mb`? The word `total_memory` is a little confusing to me.